examples/rocket_example/src/main.rs

```diff
@@ -0,0 +1,126 @@
+#[macro_use]
+extern crate rocket;
+
+use rocket::fairing::{self, AdHoc};
+use rocket::form::{Context, Form};
+use rocket::fs::{relative, FileServer};
+use rocket::request::FlashMessage;
+use rocket::response::{Flash, Redirect};
+use rocket::serde::json::Json;
+use rocket::{Build, Request, Rocket};
+use rocket_db_pools::{sqlx, Connection, Database};
+use rocket_dyn_templates::{context, Template};
+
+use sea_orm::entity::*;
+use sea_orm::RocketDbPool;
+
+mod setup;
+
+#[derive(Database, Debug)]
+#[database("rocket_example")]
+struct Db(RocketDbPool);
+
+type Result<T, E = rocket::response::Debug<sqlx::Error>> = std::result::Result<T, E>;
+
+mod post;
+pub use post::Entity as Post;
+
+#[get("/new")]
+fn new() -> Template {
+    Template::render("new", &Context::default())
+}
+
+#[post("/", data = "<post_form>")]
+async fn create(conn: Connection<Db>, post_form: Form<post::Model>) -> Flash<Redirect> {
+    let post = post_form.into_inner();
+
+    let _post = post::ActiveModel {
+        title: Set(post.title.to_owned()),
+        text: Set(post.text.to_owned()),
+        ..Default::default()
+    }
+    .save(&conn)
+    .await
+    .expect("could not insert post");
+
+    Flash::success(Redirect::to("/"), "Post successfully added.")
+}
+
+#[get("/")]
+async fn list(conn: Connection<Db>, flash: Option<FlashMessage<'_>>) -> Template {
+    let posts = Post::find()
+        .all(&conn)
+        .await
+        .expect("could not retrieve posts")
+        .into_iter()
+        .collect::<Vec<_>>();
+    let flash = flash.map(FlashMessage::into_inner);
+
+    Template::render(
+        "index",
+        context! {
+            posts: posts,
+            flash: flash,
+        },
+    )
+}
+
+#[get("/<id>")]
+async fn read(conn: Connection<Db>, id: i64) -> Option<Json<post::Model>> {
+    let post: Option<post::Model> = Post::find_by_id(id)
+        .one(&conn)
+        .await
+        .expect("could not find post");
```

Comment:
Will this panic be handled by rocket or it will halt the entire server?
Otherwise, if the post is not found, we should return a `404`.

Reply:
Oh I tried, it is 404 indeed.